src/Maestro/Client/src/MaestroApi.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Maestro.Client
+{
+    partial class MaestroApiResponseClassifier
+    {
+        public override bool IsRetriableException(Exception exception)
+        {
+            return base.IsRetriableException(exception);
+        }
+    }
+
+    partial class MaestroApi
+    {
+        partial void HandleFailedRequest(RestApiException ex)
+        {
+            if (ex.Response.Status == (int)HttpStatusCode.BadRequest)
```

Comment:
What is this method supposed to do? There are a lot of silent returns in here. That seems bad.

Reply:
this method is trying to implement the methods from the MaestroApi generated code.

Reply:
This method is intended to be any special error handling the consumer of the generated code can do. If this method returns without throwing a specific exception then the generic RestApiException is thrown.

Reply:
Ah. So it has build in handling... Makes sense with the way partials work.  So the early returns are normal.  I wonder if a comment in here somewhere might help future code lookers not get concerned like I did. :-)

Reply:
Yea, each generated api method has calls to 3 partial methods in the failure path, one HandleFailed{MethodName}Request(), one HandleFailedRequest() on the specific api class, and then one HandleFailedRequest() on the main api class. These allow specific error handling for any possible case.

What method name would you suggest? I couldn't think of anything that made sense.

Reply:
Yeah... it's a sort of... unusual concept.  I don't have a good verb on the tip of my tounge.  PreHandleFailedRequest? OptionalHandeFailedRequest?  None of them is really better than what we have.  I'd probably just plop some comments in this implementation saying "we don't have anything better to do, return to the caller and let them throw a generic exception".

Reply:
added comments to address this confusion.